migrate/list-convert.go

```
@@ -0,0 +1,296 @@
+	}
+
+	// Then we list the drives
+	// This is getting all the info from all the drives
+	ctx, _ := context.WithCancel(context.Background())
+	driveList, err := driveInterface.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		fmt.Println("error")
+		os.Exit(1)
+	}
+	volumeList, err := volumeInterface.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		fmt.Println("error")
+		os.Exit(1)
+	}
+
+	// Create volume list
+	filteredVolumes := []directcsi.DirectCSIVolume{}
+	for _, volume := range volumeList.Items {
+		// We currently don't filter volumes
+		filteredVolumes = append(filteredVolumes, volume)
+	}
+
+	// To filter the drives by status and version:
+	filteredDrives := []directcsi.DirectCSIDrive{}
+	for _, drive := range driveList.Items {
+		isDriveReadyOrInUse := false
+		isDrivev1beta5Version := false
+		isFilesystemUUIDPopulated := false
+		isPathInDevNotation := false
+		isDriveFinalizerCorrect := false
+		// Only for Ready and InUse status are valid here for this list of migration:
+		if drive.Status.DriveStatus == directcsi.DriveStatusReady || drive.Status.DriveStatus == directcsi.DriveStatusInUse {
+			isDriveReadyOrInUse = true
+		}
+		// Only for version v1beta5 is applicable the migration
+		groupAndVersion := strings.Split(drive.TypeMeta.APIVersion, `/`)
+		gvk, _ := directcsiclient.GetGroupKindVersions(
+			groupAndVersion[0],
+			drive.TypeMeta.Kind,
+			groupAndVersion[1],
+			directv1beta4.Version,
+			directv1beta3.Version,
+			directv1beta2.Version,
+			directv1beta1.Version,
+			directv1alpha1.Version,
+		)
+		if gvk.Group == "direct.csi.min.io" && gvk.Version == "v1beta5" {
+			isDrivev1beta5Version = true
+		} else {
+			isDrivev1beta5Version = false
+			fmt.Println("This drive's version cannot be migrated: ", gvk.Group, gvk.Version)
+		}
+		// Validate that FSUUID is not empty for Ready and InUse Drives
+		if len(drive.Status.FilesystemUUID) > 0 {
+			isFilesystemUUIDPopulated = true
+		} else {
+			isFilesystemUUIDPopulated = false
```

We should fail the migration if the fields are invalid instead of logging and moving forward